android/gcm/gcmsender/src/main/java/gcm/play/android/samples/com/gcmsender/GcmSender.java

```diff
@@ -26,6 +26,12 @@
 import java.net.HttpURLConnection;
 import java.net.URL;
 
+// NOTE:
+// This class emulates a server for the purposes of this sample,
+// but it's not meant to serve as an example for a production app server.
+// This class should also not be included in the client (Android) applicaiton
+// since it includes the server's API key. For information on GCM server
+// implementaion see: https://developers.google.com/cloud-messaging/server
 public class GcmSender {
```

Comment:
Do you think it would be too extreme to mark this with @Deprecated?  This way people who try to use the class will get an IDE warning.